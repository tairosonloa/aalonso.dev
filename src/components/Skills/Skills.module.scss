@import 'src/styles/utils.scss';
@import 'src/styles/palette.scss';

.container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: grid_units(17) grid_units(4);
  background-color: $white;
  text-align: center;
}

.header {
  padding-bottom: grid_units(8);
}

.skillsContainer {
  display: grid;
  grid-gap: grid_units(15);
  grid-template-columns: repeat(auto-fit, minmax(grid_units(50), 1fr));
  align-items: start;
  width: 100%;
  max-width: grid_units(200);
  padding-right: grid_units(4);
  padding-left: grid_units(4);
  text-align: center;
}

.headIcon {
  color: $burgundy;
  font-size: 40px;
}

.skillHeaderText {
  margin-top: grid_units(2);
  margin-bottom: grid_units(4);
}

.skillsLogos {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  width: 100%;

  & > div {
    margin: grid_units(2);
  }
}

.skillIcon {
  margin-bottom: grid_units(1);
  color: $burgundy;
  font-size: 35px;
}

.iconWithDescription {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}
