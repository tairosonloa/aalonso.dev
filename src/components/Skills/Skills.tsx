import React, { FC } from 'react'
import { DevIcon, DevIconName } from '../DevIcon/DevIcon'
import { Icon, IconName } from '../Icon/Icon'
import { Text, TextType } from '../Text/Text'
import styles from './Skills.module.scss'

const techSkills = [
  { icon: DevIconName.REACT, name: 'React' },
  { icon: DevIconName.NODE_JS, name: 'Node.js' },
  { icon: DevIconName.TYPESCRIPT, name: 'TypeScript' },
  { icon: DevIconName.HTML, name: 'HTML 5' },
  { icon: DevIconName.CSS, name: 'CSS 3' },
  { icon: DevIconName.PYTHON, name: 'Python 3' },
  { icon: DevIconName.POSTGRESQL, name: 'PostgreSQL' },
  { icon: DevIconName.MONGODB, name: 'MongoDB' },
  { icon: DevIconName.LINUX, name: 'Linux' },
  { icon: DevIconName.DOCKER, name: 'Docker' },
  { icon: DevIconName.AWS, name: 'Amazon AWS' },
]

const softSkills = [
  { icon: IconName.TEAM_WORK, name: 'Teamwork' },
  { icon: IconName.COMMUNICATION, name: 'Communication' },
  { icon: IconName.EMPATHY, name: 'Empathy' },
  { icon: IconName.CURIOSITY, name: 'Curiosity' },
  { icon: IconName.TIME_MANAGEMENT, name: 'Time management' },
  { icon: IconName.PROBLEM_SOLVING, name: 'Problem solving' },
  { icon: IconName.ACCOUNTABILITY, name: 'Accountability' },
  { icon: IconName.OPEN_MINDEDNESS, name: 'Open-mindedness' },
  { icon: IconName.CREATIVITY, name: 'Creativity' },
  { icon: IconName.ADAPTABILITY, name: 'Adaptability' },
  { icon: IconName.PROACTIVITY, name: 'Proactivity' },
]

export const Skills: FC = () => {
  return (
    <div className={styles.container}>
      <div className={styles.header}>
        <Text textType={TextType.TITLE}>What I offer</Text>
        <Text textType={TextType.TITLE_SMALL}>Why you might want to work with me.</Text>
      </div>
      <div className={styles.skillsContainer}>
        <div data-testid="tech-skills">
          <Icon name={IconName.TECH_SKILLS} className={styles.headIcon} />
          <Text textType={TextType.PARAGRAPH_LEAD} className={styles.skillHeaderText}>
            Tech Skills
          </Text>
          <div className={styles.skillsLogos}>
            {techSkills.map((skill) => (
              <div className={styles.iconWithDescription} key={skill.name}>
                <DevIcon name={skill.icon} className={styles.skillIcon} />
                <span>{skill.name}</span>
              </div>
            ))}
          </div>
        </div>
        <div data-testid="soft-skills">
          <Icon name={IconName.SOFT_SKILLS} className={styles.headIcon} />
          <Text textType={TextType.PARAGRAPH_LEAD} className={styles.skillHeaderText}>
            Soft Skills
          </Text>
          <div className={styles.skillsLogos}>
            {softSkills.map((skill) => (
              <div className={styles.iconWithDescription} key={skill.name}>
                <Icon name={skill.icon} className={styles.skillIcon} />
                <span>{skill.name}</span>
              </div>
            ))}
          </div>
        </div>
      </div>
    </div>
  )
}
