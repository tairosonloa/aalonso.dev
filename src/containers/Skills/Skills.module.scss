@import 'src/styles/utils.scss';
@import 'src/styles/palette.scss';

.skillsContainer {
  display: grid;
  grid-gap: grid_units(15);
  grid-template-columns: repeat(auto-fit, minmax(grid_units(76), 1fr));
  align-items: start;
  width: 100%;
  max-width: grid_units(200);
  padding-right: grid_units(4);
  padding-left: grid_units(4);
  text-align: center;

  @include screenIn(mobile) {
    grid-template-columns: repeat(auto-fit, minmax(grid_units(56), 1fr));
  }
}

.headIcon {
  color: $burgundy;
  font-size: 40px;
}

.skillHeaderText {
  margin-top: grid_units(2);
  margin-bottom: grid_units(4);
}

.skillsLogos {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  width: 100%;
  max-width: grid_units(100);
  margin: auto;

  & > div {
    margin: grid_units(2);
  }
}

.skillIcon {
  margin-bottom: grid_units(1);
  color: $burgundy;
  font-size: 35px;
}

.iconWithDescription {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}
