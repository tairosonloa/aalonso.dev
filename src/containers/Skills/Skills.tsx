import React, { FC } from 'react'
import { SectionContainer } from '../../components/DataDisplays/SectionContainer/SectionContainer'
import { SectionHeader } from '../../components/DataDisplays/SectionHeader/SectionHeader'
import { Text, TextType } from '../../components/DataDisplays/Text/Text'
import { DevIcon, DevIconName } from '../../components/Icons/DevIcon/DevIcon'
import { IconName, RemixIcon } from '../../components/Icons/RemixIcon/RemixIcon'
import styles from './Skills.module.scss'

const techSkills = [
  { icon: DevIconName.REACT, name: 'React' },
  { icon: DevIconName.NODE_JS, name: 'Node.js' },
  { icon: DevIconName.TYPESCRIPT, name: 'TypeScript' },
  { icon: DevIconName.HTML, name: 'HTML 5' },
  { icon: DevIconName.CSS, name: 'CSS 3' },
  { icon: DevIconName.PYTHON, name: 'Python 3' },
  { icon: DevIconName.POSTGRESQL, name: 'PostgreSQL' },
  { icon: DevIconName.MONGODB, name: 'MongoDB' },
  { icon: DevIconName.LINUX, name: 'Linux' },
  { icon: DevIconName.DOCKER, name: 'Docker' },
  { icon: DevIconName.AWS, name: 'Amazon AWS' },
]

const softSkills = [
  { icon: IconName.TEAM_WORK, name: 'Teamwork' },
  { icon: IconName.COMMUNICATION, name: 'Communication' },
  { icon: IconName.EMPATHY, name: 'Empathy' },
  { icon: IconName.CURIOSITY, name: 'Curiosity' },
  { icon: IconName.TIME_MANAGEMENT, name: 'Time management' },
  { icon: IconName.PROBLEM_SOLVING, name: 'Problem solving' },
  { icon: IconName.ACCOUNTABILITY, name: 'Accountability' },
  { icon: IconName.OPEN_MINDEDNESS, name: 'Open-mindedness' },
  { icon: IconName.CREATIVITY, name: 'Creativity' },
  { icon: IconName.ADAPTABILITY, name: 'Adaptability' },
  { icon: IconName.PROACTIVITY, name: 'Proactivity' },
]

export const Skills: FC = () => {
  return (
    <SectionContainer>
      <SectionHeader title="What I offer" subtitle="Why you might want to work with me." />
      <div className={styles.skillsContainer}>
        <div data-testid="tech-skills">
          <RemixIcon name={IconName.TECH_SKILLS} className={styles.headIcon} />
          <Text textType={TextType.PARAGRAPH_LEAD} className={styles.skillHeaderText}>
            Tech Skills
          </Text>
          <div className={styles.skillsLogos}>
            {techSkills.map((skill) => (
              <div className={styles.iconWithDescription} key={skill.name}>
                <DevIcon name={skill.icon} className={styles.skillIcon} />
                <span>{skill.name}</span>
              </div>
            ))}
          </div>
        </div>
        <div data-testid="soft-skills">
          <RemixIcon name={IconName.SOFT_SKILLS} className={styles.headIcon} />
          <Text textType={TextType.PARAGRAPH_LEAD} className={styles.skillHeaderText}>
            Soft Skills
          </Text>
          <div className={styles.skillsLogos}>
            {softSkills.map((skill) => (
              <div className={styles.iconWithDescription} key={skill.name}>
                <RemixIcon name={skill.icon} className={styles.skillIcon} />
                <span>{skill.name}</span>
              </div>
            ))}
          </div>
        </div>
      </div>
    </SectionContainer>
  )
}
