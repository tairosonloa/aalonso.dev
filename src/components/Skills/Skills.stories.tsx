import React from 'react'
import { dimensionsDecorator } from '../../../.storybook/utils/Dimensions'
import { Skills } from './Skills'

export default {
  title: 'Common|Skills',
  component: Skills,
  decorators: [dimensionsDecorator()],
}

export const Default = (): JSX.Element => <Skills />
