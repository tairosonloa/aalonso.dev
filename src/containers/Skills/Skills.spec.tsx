import { render } from '@testing-library/react'
import * as React from 'react'
import { Skills } from './Skills'

describe('<Skills/>', () => {
  it('should display the default message', () => {
    const title = 'What I offer'
    const subtitle = 'Why you might want to work with me.'

    const renderResult = render(<Skills />)

    expect(renderResult.getByText(title)).toBeTruthy()
    expect(renderResult.getByText(subtitle)).toBeTruthy()
  })

  it('should match the snapshot', () => {
    const renderResult = render(<Skills />)
    expect(renderResult).toMatchSnapshot()
  })
})
