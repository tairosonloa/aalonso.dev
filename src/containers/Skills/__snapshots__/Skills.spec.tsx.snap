// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skills/> should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container light"
        id="skills"
      >
        <div
          class="container hidden"
        >
          <h2
            class="common title"
            data-testid="section-title"
          >
            What I offer
          </h2>
          <h3
            class="common titleSmall"
            data-testid="section-subtitle"
          >
            What people get when work with me.
          </h3>
        </div>
        <div
          class="skillsContainer"
        >
          <div
            class="hidden"
            data-testid="tech-skills"
          >
            <h4
              class="common paragraphLead skillHeaderText"
            >
              Tech Skills
            </h4>
            <div
              class="skillsLogos"
            >
              <div
                class="iconWithDescription"
              >
                <span>
                  React
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  Node.js
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  TypeScript
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  HTML 5
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  CSS 3
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  Python 3
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  PostgreSQL
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  MongoDB
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  Linux
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  Docker
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  Amazon AWS
                </span>
              </div>
            </div>
          </div>
          <div
            class="hidden"
            data-testid="soft-skills"
          >
            <h4
              class="common paragraphLead skillHeaderText"
            >
              Soft Skills
            </h4>
            <div
              class="skillsLogos"
            >
              <div
                class="iconWithDescription"
              >
                <span>
                  Teamwork
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  Communication
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  Empathy
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  Curiosity
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  Time management
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  Problem solving
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  Accountability
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  Open-mindedness
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  Creativity
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  Adaptability
                </span>
              </div>
              <div
                class="iconWithDescription"
              >
                <span>
                  Proactivity
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container light"
      id="skills"
    >
      <div
        class="container hidden"
      >
        <h2
          class="common title"
          data-testid="section-title"
        >
          What I offer
        </h2>
        <h3
          class="common titleSmall"
          data-testid="section-subtitle"
        >
          What people get when work with me.
        </h3>
      </div>
      <div
        class="skillsContainer"
      >
        <div
          class="hidden"
          data-testid="tech-skills"
        >
          <h4
            class="common paragraphLead skillHeaderText"
          >
            Tech Skills
          </h4>
          <div
            class="skillsLogos"
          >
            <div
              class="iconWithDescription"
            >
              <span>
                React
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                Node.js
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                TypeScript
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                HTML 5
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                CSS 3
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                Python 3
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                PostgreSQL
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                MongoDB
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                Linux
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                Docker
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                Amazon AWS
              </span>
            </div>
          </div>
        </div>
        <div
          class="hidden"
          data-testid="soft-skills"
        >
          <h4
            class="common paragraphLead skillHeaderText"
          >
            Soft Skills
          </h4>
          <div
            class="skillsLogos"
          >
            <div
              class="iconWithDescription"
            >
              <span>
                Teamwork
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                Communication
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                Empathy
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                Curiosity
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                Time management
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                Problem solving
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                Accountability
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                Open-mindedness
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                Creativity
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                Adaptability
              </span>
            </div>
            <div
              class="iconWithDescription"
            >
              <span>
                Proactivity
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
